docs/guides/v8-upgrade-guide.md

```diff
@@ -39,10 +39,46 @@ You can find the legacy v7 documentation in the dropdown in the left sidebar.
 
 ### Removal of deprecated components and properties
 
-One of the changes of v8.0. is that we deleted all components and properties marked as `deprecated` in v7. Since these are not backwards compatible anymore, please make sure to update everything.
+One of the changes of v8.0 is that we deleted all components and properties marked as `deprecated` in v7. Since these are not backwards compatible anymore, please make sure to update everything.
 
 **We made a list of all the affected components and props on [Deprecated Properties and Components in Version 7.0](https://instructure.design/v7/#v7-deprecated-props-and-components) page to have a clear overview of the changes, removals, substitutes.** You can also find these changes in our [CHANGELOG](#CHANGELOG).
 
+### Upgrade step-by-step
+
+We strongly recommend to use codemods to ease your update process. To run these please install `jscodeshift` (v0.13 or newer) to your terminal. You can push your code after each step.
+
+#### 1. Run the v7 `updatePropNames` codemod:
+
+```sh
+jscodeshift -t node_modules/@instructure/ui-codemods/lib/updatePropNames.js <path> --config=node_modules/@instructure/instui-config/codemod-configs/v7/propNames.config.json
+```
+
+This will do most of the simple upgrades (renaming props, deleting props).
+
+#### 2. Run the `updateV7Props` codemod:
+
+```sh
+jscodeshift -t node_modules/@instructure/ui-codemods/lib/updateV7Props.js <path>
+```
+
+This will update more complex components (e.g. Buttons), and will tell you if it needs a manual update in the terminal. Fix what is needed manually.
+
+#### 3. Update your dependencies latest InstUI 8 and then run
+
+```sh
+jscodeshift -t node_modules/@instructure/ui-codemods/lib/updateV8Props.js <path>
+```
+
+This will update props whose name changed in v7-v8 update (there is just one, we renamed `theme` to `themeOverride`). Also run
+
+```sh
+jscodeshift -t node_modules/@instructure/ui-codemods/lib/updateImports.js <path> --config=node_modules/@instructure/instui-config/codemod-configs/v8/imports.config.js
```

Comment:
this `imports.config.js` was under the v7 folder, but it fit much more here, since its only needed when you update your dependencies to v8

Reply:
> Edit: I can see that you moved the propNames.config.js from the v8 to v7, isn't it what you meant?

yes, thats the change